File: JsonSchemaAndStructs/KiteConnect-Kotlin-Structs/gtt_delete_order.kt
// To parse the JSON, install Klaxon and do:
//
//   val gttDeleteOrder = GttDeleteOrder.fromJson(jsonString)

package quicktype

import com.beust.klaxon.*

private val klaxon = Klaxon()

data class GttDeleteOrder (
    @Json(name = "\$ref")
    val ref: String,

    @Json(name = "\$schema")
    val schema: String,

    val definitions: Definitions
) {
    public fun toJson() = klaxon.toJsonString(this)

    companion object {
        public fun fromJson(json: String) = klaxon.parse<GttDeleteOrder>(json)
    }
}

data class Definitions (
    @Json(name = "Data")
    val data: Data,

    @Json(name = "GttDeleteOrder")
    val gttDeleteOrder: GttDeleteOrderClass
)

data class Data (
    val additionalProperties: Boolean,
    val properties: DataProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class DataProperties (
    @Json(name = "trigger_id")
    val triggerID: TriggerID
)

data class TriggerID (
    val type: String
)

data class GttDeleteOrderClass (
    val additionalProperties: Boolean,
    val properties: GttDeleteOrderProperties,
    val required: List<String>,
    val title: String,
    val type: String
)

data class GttDeleteOrderProperties (
    val data: DataClass,
    val status: TriggerID
)

data class DataClass (
    @Json(name = "\$ref")
    val ref: String
)

File: JsonSchemaAndStructs/KiteConnect-Dart-Structs/gtt_delete_order.dart
// To parse this JSON data, do
//
//     final gttDeleteOrder = gttDeleteOrderFromMap(jsonString);

import 'package:freezed_annotation/freezed_annotation.dart';
import 'dart:convert';

part 'gtt_delete_order.freezed.dart';
part 'gtt_delete_order.g.dart';

@freezed
abstract class GttDeleteOrder with _$GttDeleteOrder {
    const factory GttDeleteOrder({
        Data data,
        String status,
    }) = _GttDeleteOrder;

    factory GttDeleteOrder.fromJson(Map<String, dynamic> json) => _$GttDeleteOrderFromJson(json);
}

@freezed
abstract class Data with _$Data {
    const factory Data({
        int triggerId,
    }) = _Data;

    factory Data.fromJson(Map<String, dynamic> json) => _$DataFromJson(json);
}
